AbletonMCP_Remote_Script/handlers/session.py

```
@@ -0,0 +1,1110 @@
+def set_song_settings(song, signature_numerator=None, signature_denominator=None,
+                       swing_amount=None, clip_trigger_quantization=None,
+                       midi_recording_quantization=None, back_to_arranger=None,
+                       follow_song=None, draw_mode=None,
+                       session_automation_record=None, ctrl=None):
+    """Set global song settings."""
+    try:
+        changes = {}
+        if signature_numerator is not None:
+            val = int(signature_numerator)
+            if val < 1 or val > 99:
+                raise ValueError("signature_numerator must be 1-99, got {0}".format(val))
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)